gradle-jdks/src/test/groovy/com/palantir/gradle/jdks/ConfigurationCacheTest.groovy

```diff
@@ -0,0 +1,61 @@
+/*
+ * (c) Copyright 2025 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.jdks
+
+import nebula.test.IntegrationTestKitSpec
+
+
+class ConfigurationCacheTest extends IntegrationTestKitSpec {
+
+    def setup() {
+        definePluginOutsideOfPluginBlock = true
+        keepFiles = true
+    }
+
+    def "applying PalantirCaPlugin works with configuration cache"() {
+        given:
+        // language=Gradle
+        buildFile << '''
+            apply plugin: 'com.palantir.jdks.palantir-ca'
+
+            jdks {
+                daemonTarget = 21
+            }
+
+            def value = jdks.getCaCerts().get()
```

Comment:
As a note, the lazy + interconnected nature of Gradle plugins makes it difficult to test for regressions.

e.g. In the real world, `jdks.caCerts` is probably read by some other plugin. When `jdk.caCerts` is realised, it calls the external process, violating CC requirements in the config phase. 

Without manually tracing the plugin code and triggering CC violations, we have no idea if we're hitting all paths.